source/common/http/conn_manager_impl.cc

```diff
@@ -404,6 +423,15 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
   // Set the trusted address for the connection by taking the last address in XFF.
   downstream_address_ = Utility::getLastAddressFromXFF(*request_headers_);
   decodeHeaders(nullptr, *request_headers_, end_stream);
+
+  Tracing::Decision tracing_decision =
```

Comment:
You won't, but this part should definitely be done prior to decodeHeaders(). It's possible that you might still need to do some logic/filtering when you finish the span.